test/cuda/cuda.jl

```
@@ -40,8 +40,10 @@ end
 
 @testset "onecold gpu" begin
   y = Flux.onehotbatch(ones(3), 1:10) |> gpu;
+  l = ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j']
   @test Flux.onecold(y) isa CuArray
   @test y[3,:] isa CuArray
+  @test Flux.onecold(y, l) == ['a', 'a', 'a']
```

this behavior is a bit weird right? maybe we should just error out on non-isbits labels and CuArrays?

---

Yeah a little odd since `y` is a `CuArray`, but `l` is an `Array` so not too weird. I've encountered this use case before, and my preference would be to get back a result than an error. As a user, I'd expect the first thing people try is to do `l |> gpu` or `onecold(y, l) |> gpu` if they want a `CuArray`. Both of which should throw an error, so the issue should be obvious.

---

The reason this fix is in there is to address stuff like #864